AutoRest/TestServer/swagger/azure-special-properties.json

```diff
@@ -732,6 +732,43 @@
           }
         }
       }
+    },    
+    "/azurespecials/customNamedRequestId": {
+      "post": {
+        "operationId": "header_customNamedRequestId",
+        "description": "Send foo-client-request-id = 9C4D50EE-2D56-4CD3-8152-34347DC9F2B0 in the header of the request",
+        "parameters": [
+          {
+            "name": "foo-client-request-id",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "description": "The fooRequestId",
+            "x-ms-client-request-id": { "value": "true" },
```

Comment:
forgive me if i am out of context, but any reason we want to use a json objectl rather just a bool value
     "x-ms-client-request-id": true